physcraper/__init__.py

```diff
@@ -1809,186 +1810,168 @@ def run_blast_wrapper(self):  # TODO Should this be happening elsewhere?
+        staxids_l = staxids.split(";")
+        sscinames_l = sscinames.split(";")
+        sallseqid_l = sallseqid.split(";")
+        salltitles_l = salltitles.split("<>")
+
+        # make sure you have the correct infos
+        tax_id_l = self.get_taxid_from_acc(gb_acc)
+        # debug([staxids, gb_acc, tax_id_l])
+        for item in tax_id_l:
+            try:
+                assert str(item) in staxids_l, (item, staxids_l, tax_id_l, gb_acc)
+            except AssertionError:
+                sys.stderr.write("Genbank accession {} does map to different taxonid in database: {} vs {}. Check with ncbi\n".format(gb_acc, item, staxids_l)) 
+
+        # this while loop is here to speed up the process of finding the correct information
+        count = 0
+        spn_range = 0
+        stop_while = False
+        id_before = 0
+        id_now = -2
+        while len(found_taxids) < len(set(staxids_l)):  # as long as i have not found all taxids for the seq
+            count += 1
+            if stop_while:
+                # debug("stop while")
+                break
+            if count == 5:
+                break  # too many tries to find correct number of redundant taxa
+            elif count == 1:
+                for i in range(0, len(sallseqid_l)):
+                    if len(found_taxids) == len(staxids_l):  # if we found all taxon_ids present in the initial list, stop looking for more
+                        break
+                    # debug("do all the stuff")
+                    gb_acc = get_acc_from_blast(sallseqid_l[i])
+                    gi_id = get_gi_from_blast(sallseqid_l[i])
+                    stitle = salltitles_l[i]
+                    # if both var are the same, we do not need to search GB for taxon info
+                    # staxids = tax_id_l[i]
+                    print("check out what goes wrong here")
+                    print(gb_acc, self.get_taxid_from_acc(gb_acc))
+                    print(gb_acc, self.get_taxid_from_acc(gb_acc)[0])
+
+                    staxids = int(self.get_taxid_from_acc(gb_acc)[0])  # corresponding taxid is always in first position
+                    # debug(qtaxid)
+                    id_before = id_now
+                    id_now = staxids
+          
+                    for item in tax_id_l:
+                        try:
+                            assert str(staxids) in staxids_l or str(staxids) in tax_id_l, (item, staxids_l, tax_id_l, gb_acc)
+                        except AssertionError:
+                            sys.stderr.write("Merged Genbank accession {} does map to different taxonid in database: {} vs {}. Check with ncbi\n".format(gb_acc, item, staxids_l)) 
+                    # assert str(staxids) in staxids_l or staxids in tax_id_l, (str(staxids), staxids_l, tax_id_l)
+                    
+                    # if gb acc was already read in before stop the for loop
+                    if gb_acc in query_dict or gb_acc in self.data.gb_dict:
```

Comment:
For example, if you have a function that returns a gb_acc and a taxon_id (or whatever other information is required) for each sequence id, and separate function  adds that information into the query dict for each accession number? That would have the advantage of being a function that could be reused for adding information into the query dict for unmerged sequences as well.